Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -787,15 +787,7 @@ void RenderGrid::placeItemsOnGrid(std::optional<LayoutUnit> availableLogicalWidt
 
         // FIXME: We should be able to short circuit the initial placement (https://bugs.webkit.org/show_bug.cgi?id=248291)
         performAutoPlacement();
-
-        HashMap<RenderBox*, GridArea> firstTrackItems;
-        Vector<RenderBox*> itemsWithDefiniteGridAxisPosition;
-        Vector<RenderBox*> itemsWithIndefinitePosition;
-
-        collectMasonryItems(currentGrid(), gridAxisTracksBeforeAutoPlacement, masonryAxisDirection, firstTrackItems, itemsWithDefiniteGridAxisPosition, itemsWithIndefinitePosition);
-        currentGrid().setupGridForMasonryLayout();
-        populateExplicitGridAndOrderIterator();
-        m_masonryLayout.performMasonryPlacement(firstTrackItems, itemsWithDefiniteGridAxisPosition, itemsWithIndefinitePosition, masonryAxisDirection);
+        m_masonryLayout.performMasonryPlacement(gridAxisTracksBeforeAutoPlacement, masonryAxisDirection);
```

Comment:
Can we get rid of the masonry lambda here as discussed yesterday?

```CPP
performAutoPlacement();

if(isMasonryRows(style() || isMasonryColumns(style()))
{
  // Do the masonry stuff here
}
  

```

Reply:
I tried to get this done but it looks like I can't completely do it yet because we are reliant on `unsigned gridAxisTracksBeforeAutoPlacement = currentGrid().numTracks(gridAxisDirection);` being computed before auto placement. I tried to see if using `GridPositionsResolver::explicitGridRowCount`/GridPositionsResolver::explicitGridColumnCount, but it doesn't work in all scenarios (e.g. masonry-item-placement-002). After talking with Matt, we shouldn't need this variable anyways after we implement the logic in the implicit grid section (particularly the part about ignoring `grid-auto-flow`), so maybe we can do it then

Reply:
Thanks for taking a look at this.

Not a huge deal for this patch.